java/flight/flight-core/pom.xml

```diff
@@ -151,13 +150,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <!--
-          Downgrade maven-shade-plugin specifically for this module.
-          Using a newer version up to at least 3.5.1 will cause
-          issues in the arrow-tools tests looking up FlatBuffer
-          dependencies.
-         -->
-        <version>3.2.4</version>
```

Comment:
Hmm, why does this not apply anymore? We saw problems last time we tried to upgrade

Reply:
Yes, [as discussed in the issue](https://github.com/apache/arrow/issues/41571#issuecomment-2123679316) associated to the rollback, the problem was caused by the combination of arrow configuration replaced the main artifact and its associated pom with the shaded artifact and its reduced pom.xml, and the newer versions of the plugins fixing an issue where the reduced pom.xml was not attached to the artifact.

It is solved by disabling the reduced pom.xml generation and attachment using the `createDependencyReducedPom` configuration property